controller/src/controller/models/domain/rspimage.py

```diff
@@ -549,3 +553,59 @@ def _resolve_duplicate(self, new: RSPImage, old: RSPImage) -> None:
                     other.aliases.add(new.tag)
             old.aliases.add(new.tag)
             self._by_digest[new.digest] = new
+
+    def filter(
+        self, policy: RSPImageFilterPolicy, age_basis: datetime
+    ) -> Self:
```

Comment:
Given how `filter` is used, I think it should just return `list[RSPImageTag]` or `list[RSPImage]`. None of the callers use the collection properties; they just immediately turn this into a list, so turning it into a collection and then back into the same list is unnecessary work.